src/open_samus_returns_rando/lua_editor.py

```diff
@@ -213,6 +214,13 @@ def _create_custom_init(self, editor: PatcherEditor, configuration: dict) -> str
         # use _MAX if main is unlocked to unlock the ammo too
         if "ITEM_WEAPON_MISSILE_LAUNCHER" in inventory and "ITEM_MISSILE_TANKS" in inventory:
             inventory["ITEM_WEAPON_MISSILE_MAX"] = inventory.pop("ITEM_MISSILE_TANKS")
+        else:
+            # For the gun component to work without launcher, any value > 0 is sufficient
+            inventory["ITEM_MISSILE_CHECK"] = 1
```

Comment:
You are missing the case where you start with the Missile Launcher + Ammo but haven't collected a tank in-between.
Like in the oter comment: I would suggest to just always use `ITEM_MISSILE_CHECK` and use 
```
"ITEM_MISSILE_CHECK: max(1, inventory.get("ITEM_WEAPON_MISSILE_MAX, 0))
``` 
in the `final_inventory` here.

Reply:
Im not sure what you mean by this. If you start with launcher this else block is ignored.

Reply:
Whoops yeah. Got confused. The mentoned thing would only be valid if we just always use `ITEM_MISSILE_CHECK`.